src/odfesqlodbc/results.c

```diff
@@ -1388,6 +1390,17 @@ RETCODE SQL_API ESAPI_ExtendedFetch(HSTMT hstmt, SQLUSMALLINT fFetchType,
      */
     if (!should_set_rowset_start)
         rowset_start = SC_get_rowset_start(stmt);
+
+    // Get more results when cursor reaches end 
+    if ((!useCursor) && (rowset_start >= num_tuples) && (res->server_cursor_id != NULL)) {
```

Comment:
Shouldn't it be useCursor instead of !useCursor?

Reply:
Actually I don't check for `useCursor` at all since `EXTFETCH_RETURN_EOF` can be reached in both cases. Thanks for noticing.